github/repos_releases_test.go

```diff
@@ -253,6 +253,32 @@ func TestRepositoriesService_DownloadReleaseAsset_Redirect(t *testing.T) {
 	}
 }
 
+func TestRepositoriesService_DownloadReleaseAsset_APIError(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/releases/assets/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", defaultMediaType)
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprint(w, `{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}`)
+	})
+
+	resp, loc, err := client.Repositories.DownloadReleaseAsset("o", "r", 1)
+	if err == nil {
+		t.Error("Repositories.DownloadReleaseAsset did not return an error")
+	}
+
+	if resp != nil {
+		resp.Close()
+		t.Error("Repositories.DownloadReleaseAsset returned stream, want nil")
+	}
+
+	if loc != "" {
+		t.Error(`Repositories.DownloadReleaseAsset returned "%s", want empty ""`, loc)
```

Comment:
Certainly
On Apr 21, 2016 10:17 PM, "Dmitri Shuralyov" <notifications@github.com>
wrote:

> In github/repos_releases_test.go
> <https://github.com/google/go-github/pull/344#discussion_r60692002>:
>
> > +		w.WriteHeader(http.StatusNotFound)
> > +		fmt.Fprint(w, `{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}`)
> > +	})
> > +
> > +	resp, loc, err := client.Repositories.DownloadReleaseAsset("o", "r", 1)
> > +	if err == nil {
> > +		t.Error("Repositories.DownloadReleaseAsset did not return an error")
> > +	}
> > +
> > +	if resp != nil {
> > +		resp.Close()
> > +		t.Error("Repositories.DownloadReleaseAsset returned stream, want nil")
> > +	}
> > +
> > +	if loc != "" {
> > +		t.Error(`Repositories.DownloadReleaseAsset returned "%s", want empty ""`, loc)
>
> This should be t.Errorf, not t.Error.
>
> @willnorris <https://github.com/willnorris>, will you take an improved
> .travis.yml that would run vet as part of CI? It would've caught this
> issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/google/go-github/pull/344/files/4e358c5270aa6e65264663e95b7b777c6405984a#r60692002>
>